src/poetry/console/commands/check.py

```diff
@@ -19,6 +25,32 @@ def handle(self) -> int:
         poetry_file = Factory.locate(Path.cwd())
         config = PyProjectTOML(poetry_file).poetry_config
         check_result = Factory.validate(config, strict=True)
+
+        unrecognized = sorted(
+            set(config.get("classifiers", []))
+            - set(classifiers)
+            - set(deprecated_classifiers)
+        )
+        if unrecognized:
+            check_result["errors"].append(
+                f"Unrecognized classifiers: {unrecognized!r}."
+            )
+
+        deprecated = sorted(
+            set(config.get("classifiers", [])).intersection(set(deprecated_classifiers))
+        )
+        if deprecated:
+            for old_classifier in deprecated:
+                message = f"Deprecated classifier {old_classifier!r}."
+                new_classifiers = cast(
```

Comment:
Really not a fan of this inline cast -- I think we can find a better way to write this to still satisfy Mypy.

Reply:
it's not correct either is it?  `deprecated_classifiers` is the dictionary, `new_classifiers` is just a list of strings?

Reply:
https://github.com/pypa/trove-classifiers/pull/109

Reply:
which has been merged and auto-released, so trove-classifiers 2022.6.23 should make the cast redundant

Reply:
Just removed the explicit casting and bumped `trove-classifiers` to `2022.6.23` in https://github.com/python-poetry/poetry/pull/2881/commits/404998ce080a6bd2b5d5729813d13df8061d44ea